### ISSUE TITLE
Wav2Vec2Pretrain (HFTransformersInterface implementation) samples padded values for mask_time_indices and negative_sample_indices

### ISSUE THREAD

**Issue by porfirythelaw** [speechbrain/speechbrain#2386]
### Describe the bug

I've been using SpeechBrain Wav2Vec2 training recipe (with HF integration) on my own data, and noticed that I get significantly different metrics with the same model on validation dataset depending on the amount of padding in the batch. My hypothesis was that somehow padding is not ignored during indices sampling process, and I think this is what in fact is happening.

https://github.com/speechbrain/speechbrain/blob/eba7714d4f08ffe8a0df8aed0f729b94959ca977/speechbrain/lobes/models/huggingface_transformers/wav2vec2.py#L261
```
        # 1. Compute the indices that will be masked
        mask_time_indices = _compute_mask_indices(
            (batch_size, sequence_length),
            mask_prob=self.mask_prob,
            mask_length=self.mask_length,
        )
```
As you can see in this function you don't provide attention mask, so masked indices are drawn from padded values as well.

Same for negative masked indicies, which you take from the whole sequence
https://github.com/speechbrain/speechbrain/blob/eba7714d4f08ffe8a0df8aed0f729b94959ca977/speechbrain/lobes/models/huggingface_transformers/wav2vec2.py#L278
```
        negative_sample_indices = torch.tensor(
            transformers.models.wav2vec2.modeling_wav2vec2._sample_negative_indices(
                (batch_size, sequence_length),
                num_negatives=self.config.num_negatives,
                mask_time_indices=full_sentence_indices,
            ),
            device=wav.device,
            dtype=torch.long,
        )
```

You provide attention mask in this call to the model
https://github.com/speechbrain/speechbrain/blob/eba7714d4f08ffe8a0df8aed0f729b94959ca977/speechbrain/lobes/models/huggingface_transformers/wav2vec2.py#L289
```
            self.model(
                wav,
                mask_time_indices=torch_mask_time_indices,
                sampled_negative_indices=negative_sample_indices,
                attention_mask=padding_mask,
            ),
            torch_mask_time_indices,
        )
```

However, if you check hugginface source code it does not affect loss calculation, it only affects encoder self-attention.

I'm not sure if this behavior was intended or not.

### Expected behaviour

Padded values should not be influencing model loss / metrics.

### Environment Details

Speechbrain [v0.5.16](https://github.com/speechbrain/speechbrain/releases/tag/v0.5.16)

**Comment by Adel-Moumen** [comment]
Hey @TParcollet, could you please have a look?

**Comment by TParcollet** [comment]
That's quite late to answer, but yes it certainly is true. The reason is that we rely on HF functions here. @porfirythelaw could you propose a PR with this fix? I will test it. 

Many thanks.
